src/jarabe/desktop/activitychooser.py

```diff
@@ -0,0 +1,232 @@
+        self.search_bar.connect('query-changed', self.__toolbar_query_changed_cb)
+        self.search_bar.search_entry.connect('key-press-event',
+                                             self.__key_press_event_cb)
+        self._scrolled_window = Gtk.ScrolledWindow()
+        self._scrolled_window.set_policy(Gtk.PolicyType.AUTOMATIC,
+                                         Gtk.PolicyType.AUTOMATIC)
+
+        self._scrolled_window.add(self._list_view)
+
+        self.get_vbox().pack_start(self._scrolled_window, True, True, 0)
+
+        self._list_view.show()
+        self._list_view.connect('clear-clicked',
+                                self.__activitylist_clear_clicked_cb, self.search_bar)
+
+
+        self.tree_view = self._list_view._tree_view
+
+        self.tree_view.date_column.set_visible(False)
+        self.tree_view.fav_column.set_visible(False)
+        self.tree_view.version_column.set_visible(False)
+
+        if self.tree_view.row_activated_handler:
+            self.tree_view.disconnect(self.tree_view.row_activated_handler)
+        if self.tree_view.button_press_handler:
+            self.tree_view.disconnect(self.tree_view.button_press_handler)
+        if self.tree_view.button_reslease_handler:
+            self.tree_view.disconnect(self.tree_view.button_reslease_handler)
+        if self.tree_view.icon_clicked_handler:
+            self.tree_view.disconnect(self.tree_view.icon_clicked_handler)
+
+        self.tree_view.connect('row-activated',self.__row_activated_cb)
+
+        self.show()
+
+    def __toolbar_query_changed_cb(self, toolbar, query):
+        self._query = normalize_string(query.decode('utf-8'))
+        self._list_view.set_filter(self._query)
+
+        toolbar.search_entry._icon_selected = \
+            self._list_view.get_activities_selected()
+
+        # verify if one off the selected names is a perfect match
+        # this is needed by th case of activities with names contained
+        # in other activities like 'Paint' and 'MusicPainter'
+        for activity in self._list_view.get_activities_selected():
+            if activity['name'].upper() == query.upper():
+                toolbar.search_entry._icon_selected = [activity]
+                break
+
+        # Don't change the selection if the entry has been autocompleted
+        if len(toolbar.search_entry._icon_selected) == 1 \
+           and not toolbar.search_entry.get_text() == activity['name']:
+            pos = toolbar.search_entry.get_position()
+            toolbar.search_entry.set_text(
+                toolbar.search_entry._icon_selected[0]['name'])
+            toolbar.search_entry.select_region(pos, -1)
+
+    def __key_press_event_cb(self, widget, event):
+        if not self.search_bar.search_entry.has_focus():
+            self.search_bar.search_entry.grab_focus()
+
+        if widget == self.search_bar.search_entry:
+            if event.keyval == Gdk.KEY_Return and len(widget._icon_selected) == 1:
```

Comment:
`IconEntry object has no attribute _icon_selected` -- I get the error when pressing enter.